sfdoc/publish/tests/utils.py

```diff
@@ -41,116 +50,145 @@ def create_test_html(url_name, title, summary, body):
-    query_s = urlencode({
-        'q': (
-            "SELECT {} FROM {} WHERE UrlName='{}' "
-            "AND PublishStatus='{}' AND language='en_US'"
-        ).format(
-            ','.join(fields),
-            settings.SALESFORCE_ARTICLE_TYPE,
-            url_name,
-            publish_status,
-        ),
-    })
-    url = '{}/services/data/v{}/query/?{}'.format(
-        instance_url,
-        settings.SALESFORCE_API_VERSION,
-        query_s,
+        fields = ["Id"]
+    query_s = urlencode(
+        {
+            "q": (
+                "SELECT {} FROM {} WHERE UrlName='{}' "
+                "AND PublishStatus='{}' AND language='en_US'"
+            ).format(
+                ",".join(fields),
+                settings.SALESFORCE_ARTICLE_TYPE,
+                url_name,
+                publish_status,
+            )
+        }
     )
-    responses.add(
-        'GET',
-        url=url,
-        match_querystring=True,
-        json=return_val,
+    url = "{}/services/data/v{}/query/?{}".format(
+        instance_url, settings.SALESFORCE_API_VERSION, query_s
     )
+    responses.add("GET", url=url, match_querystring=True, json=return_val)
 
 
 def mock_update_draft(instance_url, kav_id):
-    url = urljoin(instance_url, 'services/data/v{}/sobjects/{}/{}'.format(
-        settings.SALESFORCE_API_VERSION,
-        settings.SALESFORCE_ARTICLE_TYPE,
-        kav_id,
-    ))
-    responses.add(
-        'PATCH',
-        url=url,
-        status=HTTPStatus.NO_CONTENT,
+    url = urljoin(
+        instance_url,
+        "services/data/v{}/sobjects/{}/{}".format(
+            settings.SALESFORCE_API_VERSION, settings.SALESFORCE_ARTICLE_TYPE, kav_id
+        ),
+    )
+    responses.add("PATCH", url=url, status=HTTPStatus.NO_CONTENT)
+
+
+ORIGINIT = TemporaryDirectory.__init__
+
+
+@contextmanager
+def makeDebugTemporaryDirectoryMock(parent_prefix="", default_dir=""):
+    class DebugTemporaryDirectory(TemporaryDirectory):
+        def __init__(self, suffix=None, prefix="", dir=None):
+            full_prefix = parent_prefix
+            full_prefix += "_" + str(time.process_time()) + "_"
+            if getattr(self, "subprefix", None):
+                full_prefix += "_" + self.subprefix
+            if prefix:
+                full_prefix += "_" + prefix
+            ORIGINIT(self, prefix=full_prefix, dir=(dir or default_dir))
+
+        @classmethod
+        def set_subprefix(cls, subprefix):
+            print("CLASS", cls)
+            cls.subprefix = subprefix
+
+        def cleanup(self, name="", warn_message=""):
+            pass
+
+    d = DebugTemporaryDirectory
+    with mock.patch("tempfile.TemporaryDirectory.__init__", d.__init__),\
+        mock.patch("tempfile.TemporaryDirectory.cleanup", d.cleanup),\
+        mock.patch("tempfile.TemporaryDirectory._cleanup", d.cleanup): # noqa E125
+            TemporaryDirectory.set_subprefix = classmethod(d.set_subprefix.__func__) # noqa E117
+            yield TemporaryDirectory
```

Comment:
Seems like it'd be simpler to just update everywhere in the codebase that uses TemporaryDirectory to use DebugTemporaryDirectory instead.

Reply:
Do you mean updating the references with Mocking or literally changing the code?

I'm on the fence about targetted mocking but I would prefer not to introduce additional debugging stuff into the actual code. And I would have to add a feature to make it optional or else the tempdirs would not be temp anymore and Heroku's /tmp would grow for however long the dino was alive (which admittedly is not that long but still its poor practice to depend on server rebooting to prevent the temp from growing without bound).

Reply:
Oh yeah, I overlooked the part about skipping the cleanup -- it would help if this thing had a docstring explaining what it's for.

Reply:
dd83e5abc6cb1a2b759067f18a71319d1b12e93e